test/minikube/minikube_base_database_test.go

```diff
@@ -408,56 +411,91 @@ func TestKubernetesRestoreDatabase(t *testing.T) {
 			"--password", "secret",
 			"demo",
-			"--file", "/opt/nuodb/samples/quickstart/sql/create-db.sql")
+			"--file", "/opt/nuodb/samples/quickstart/sql/create-db.sql",
+		)
+
+		// verify that the database contains the populated data
+		tables, err := testlib.RunSQL(t, namespaceName, admin0, "demo", "show schema User")
+		assert.NilError(t, err, "error running SQL: show schema User")
+		assert.Check(t, strings.Contains(tables, "HOCKEY"), "tables returned: ", tables)
 
 		// run a manual backup
-		k8s.RunKubectl(t, opts,
-			"exec", admin0, "--",
-			"nuodocker", "backup", "database",
-			"--db-name", "demo",
-			"--type", "full",
-			"--timeout", "120",
-			"--backup-root", "/var/opt/nuodb/backup",
-		)
+		// k8s.RunKubectl(t, opts,
+		// 	"exec", admin0, "--",
+		// 	"nuodocker", "backup", "database",
+		// 	"--db-name", "demo",
+		// 	"--type", "full",
+		// 	"--timeout", "120",
+		// 	"--backup-root", "/var/opt/nuodb/backup",
+		// )
+
+		// // trigger a manual backup now
+		// k8s.RunKubectl(t, opts,
+		// 	"exec", admin0, "--",
+		// 	"nuocmd", "set", "value", "--key", fmt.Sprintf("%s/demo/cluster0", testlib.BACKUP_SEMAPHORE), "--value", "true", "--unconditional",
+		// )
+
+		// wait for backup to complete
+		// testlib.Await(t, func() bool {
+		// 	output, err := k8s.RunKubectlAndGetOutputE(t, opts,
+		// 		"exec", admin0, "--",
+		// 		"nuocmd", "get", "value", "--key", fmt.Sprintf("%s/demo/cluster0", testlib.BACKUP_SEMAPHORE),
+		// 	)
+		// 	assert.NilError(t, err, "Error getting value from KV")
+		// 	return output == ""
+		// }, 240*time.Second)
+
+		// // wait for the backup to complete - up to 4 mins to start, and another 4 mins to then complete...
+		// databaseName := "demo"
+		// backupJob := fmt.Sprintf("hotcopy-%s-job-initial-", databaseName)
+		// testlib.AwaitPodPhase(t, namespaceName, backupJob, corev1.PodSucceeded, 120*time.Second)
 
 		// populate some more data
-		k8s.RunKubectl(t, opts,
-			"exec", admin0, "--",
-			"/opt/nuodb/bin/nuosql",
-			"--user", "dba",
-			"--password", "secret",
-			"demo",
-			"--file", "/opt/nuodb/samples/quickstart/sql/Teams.sql",
-		)
+		// k8s.RunKubectl(t, opts,
+		// 	"exec", admin0, "--",
+		// 	"/opt/nuodb/bin/nuosql",
+		// 	"--user", "dba",
+		// 	"--password", "secret",
+		// 	"demo",
+		// 	"--file", "/opt/nuodb/samples/quickstart/sql/Teams.sql",
+		// )
 
 		defer testlib.Teardown(testlib.TEARDOWN_RESTORE)
 		restoreDatabase(t, namespaceName)
 
+		// NTJ - Now automatically triggered by the restore chart (by default - can be overridden)
 		// and restart database - to trigger the restore
-		k8s.RunKubectl(t, opts,
-			"exec", admin0, "--",
-			"nuocmd", "shutdown", "database",
-			"--db-name", "demo",
-		)
-
+		// k8s.RunKubectl(t, opts,
+		// 	"exec", admin0, "--",
+		// 	"nuocmd", "shutdown", "database",
+		// 	"--db-name", "demo",
+		// )
+
+		// wait for the datbase to be restarted (shutdown followed by restart)
+		testlib.AwaitDatabaseDown(t, namespaceName, admin0, "demo")
```

Comment:
you can not write the test like this. It will fail randomly. By the time control reaches this point, the process might be already up and this will timeout.

Reply:
I take your point - although:
* database shutdown is not quick - it takes time for the processes to shut down, and report that fact to admin.

* the database restart is triggered by `k8s` - which involves guaranteed delays (timeouts).
In addition, the SM startup performs multiple `nuocmd get value` operations, each of which are slow - a second or more elapsed to complete.
So the window for any race condition relies on the database shutting down unusually quickly, and `k8s` detecting that and restarting the TEs quickly enough that all that has occurred _before_ the `check database` command has commenced.

>  It will fail randomly. 
It *may* fail randomly...

> You have to either shut the DB down manually 
That is what is happening - norestore calls `nuocmd shutdown database`;
After which, `k8s` restarts the database pods 

> or await the incarnation in the admin
Yes - I believe this is the only guaranteed deterministic solution
I will change the test to retrieve the incarnation number, and then Await for that to increase.

Reply:
let me rephrase. This has failed and this is failing. So this needs to be fixed before it can go in.

If you wanna talk about how to fix it, we can either do it here or in SLACK.

Reply:
My apologies for being unclear.
When I wrote:
> I will change the test to retrieve the incarnation number, and then Await for that to increase.
I meant that I am making that change now.